bundles/core/org.eclipse.smarthome.core.voice.test/src/test/java/org/eclipse/smarthome/core/voice/javavoicemanager/VoiceManagerTest.java

```diff
@@ -0,0 +1,386 @@
+        hliStub = new HumanLanguageInterpreterStub();
+        source = new AudioSourceStub();
+
+        exception.expect(IllegalStateException.class);
+        voiceManager.startDialog(ksService, sttService, ttsService, hliStub, source, sink, Locale.getDefault(), "word",
+                null);
+
+        assertFalse(ksService.isWordSpotted());
+    }
+
+    @Test
+    public void startDialogWhenAllOfTheRequiredServicesAreAvailable() {
+        sttService = new STTServiceStub();
+        ksService = new KSServiceStub();
+        ttsService = new TTSServiceStub();
+        hliStub = new HumanLanguageInterpreterStub();
+        source = new AudioSourceStub();
+
+        registerService(sttService);
+        registerService(ksService);
+        registerService(ttsService);
+        registerService(hliStub);
+        registerService(source);
+
+        voiceManager.startDialog(ksService, sttService, ttsService, hliStub, source, sink, null, "word", null);
+
+        assertTrue(ksService.isWordSpotted());
+    }
+
+    @Test
+    public void startDialogAndVerifyThatAKSExceptionIsProperlyHandled() {
+        sttService = new STTServiceStub();
+        ksService = new KSServiceStub();
+        ttsService = new TTSServiceStub();
+        hliStub = new HumanLanguageInterpreterStub();
+        source = new AudioSourceStub();
+
+        registerService(sttService);
+        registerService(ksService);
+        registerService(ttsService);
+        registerService(hliStub);
+        registerService(source);
+
+        ksService.setIsKsExceptionExpected(true);
+
+        voiceManager.startDialog(ksService, sttService, ttsService, hliStub, source, sink, null, "", null);
+
+        assertFalse(ksService.isWordSpotted());
+    }
+
+    @Test
+    public void startDialogWithoutPassingAnyParameters() throws IOException, InterruptedException {
+        sttService = new STTServiceStub();
+        ksService = new KSServiceStub();
+        ttsService = new TTSServiceStub();
+        hliStub = new HumanLanguageInterpreterStub();
+        source = new AudioSourceStub();
+
+        registerService(sttService);
+        registerService(ksService);
+        registerService(ttsService);
+        registerService(hliStub);
+        registerService(source);
+
+        Dictionary<String, Object> config = new Hashtable<String, Object>();
+        config.put(CONFIG_KEYWORD, "word");
+        config.put(CONFIG_DEFAULT_STT, sttService.getId());
+        config.put(CONFIG_DEFAULT_KS, ksService.getId());
+        config.put(CONFIG_DEFAULT_HLI, hliStub.getId());
+        config.put(CONFIG_DEFAULT_VOICE, voice.getUID());
+
+        ConfigurationAdmin configAdmin = super.getService(ConfigurationAdmin.class);
+        String pid = "org.eclipse.smarthome.voice";
+        Configuration configuration = configAdmin.getConfiguration(pid);
+        configuration.update(config);
+
+        // To allow the config admin to update the configuration.
+        Thread.sleep(100);
```

Comment:
Please put the below two lines into a `waitForAssert(() -> {})`-Block instead. The fixed sleep is guaranteed to fail one day on some CI server...